numpy/lib/npyio.py

```diff
@@ -1984,9 +1988,23 @@ def genfromtxt(fname, dtype=float, comments='#', delimiter=None,
     if usemask:
         output = output.view(MaskedArray)
         output._mask = outputmask
+    # Verify that the array has at least dimensions `ndmin`.
+    # Check correctness of the values of `ndmin`
+    if ndmin not in [0, 1, 2]:
+        raise ValueError('Illegal value of ndmin keyword: %s' % ndmin)
+    # Tweak the size and shape of the arrays - remove extraneous dimensions
+    if output.ndim > ndmin:
+        output = np.squeeze(output)
+    # and ensure we have the minimum number of dimensions asked for
+    # - has to be in this order for the odd case ndmin=1, X.squeeze().ndim=0
+    if output.ndim < ndmin:
+        if ndmin == 1:
+            output = np.atleast_1d(output)
+        elif ndmin == 2:
+            output = np.atleast_2d(output).T
     if unpack:
-        return output.squeeze().T
-    return output.squeeze()
+        return output.T
+    return output
```

Comment:
Rather than copying these lines from the other function, can you extract them to a local `_squeeze_to_ndmin(arr, ndmin)` function?